lib/convection/control/cloud.rb

```diff
@@ -114,6 +114,18 @@ def diff(to_stack, options = {}, &block)
           return
         end
 
+        errors = []
+        stacks.each_value do |stack|
+          if stack.error?
+            errors << stack.errors.collect { |x| x.exception.message }
+          end
+        end
+        errors = errors.uniq.flatten!
+        unless errors.empty?
+          block.call(Model::Event.new(:error, "Error(s) during stack diff #{errors.join(', ')}", :error), errors) if block
+          return
+        end
```

Comment:
1/2. No, if we took that approach we would be doing number 2 where we only error on stacks that have errors and proceed with the diff for stacks in a good state. I do not feel like that is a good idea as there is a possibility the user will miss the error. I feel we should error if any stacks had trouble being initialized. 
3. We should not proceed with the diff if an error occurs. We do not exit with a non zero exit code. (same as other error handling for converges or deletes)